stream.go

```diff
@@ -856,7 +858,33 @@ type ServerStream interface {
 	// SetTrailer sets the trailer metadata which will be sent with the RPC status.
 	// When called more than once, all the provided metadata will be merged.
 	SetTrailer(metadata.MD)
-	Stream
+	// Context returns the context for this stream.
+	Context() context.Context
+	// SendMsg sends a message. On error, SendMsg aborts the stream and the
+	// error is returned directly.
+	//
+	// SendMsg blocks until:
+	//   - There is sufficient flow control to schedule m with the transport, or
+	//   - The stream is done, or
+	//   - The stream breaks.
+	//
+	// SendMsg does not wait until the message is received by the server. An
+	// untimely stream closure may result in lost messages. To ensure delivery,
```

Comment:
The server can never know whether the client received the message without something in the user's protocol.

Reply:
Right - this line alludes to that. I could more strongly state that by adjusting the line to:

"SendMsg does not wait until the message is received by the client. Users who need per-message acknowledgement are expected to implement it on top of gRPC."

Though, I don't know how much value there is in that. I kind of think the brevity of just saying, "we don't wait" is enough of a warning. Thoughts?

Reply:
SGTM

Reply:
Will you update this?  The bit about "ensure the RPC completed successfully using RecvMsg" is misleading/wrong on the server-side.

Reply:
Ping on this comment..

Reply:
Shoot, sorry to have missed this. Hm, good point - that just means the client has finished sending their messages. Bad copy paste. I'm not sure it's worth saying, but is it correct to say that once the server returns in the handler, all messages will have been sent successfully? Or is the promise still only that messages have been scheduled? In short, is there any way to ensure all server messages have arrived? (I think not, but want to make sure I ask)

In the meantime, I've simply removed the line about RecvMsg. I neglected to include the line about per-message acks, since it doesn't seem to add _too_ much more.

Reply:
> I'm not sure it's worth saying, but is it correct to say that once the server returns in the handler, all messages will have been sent successfully? 

No, the server never knows whether a message is received by the client unless the user builds their own acks into their protocol.  I think what you were proposing before was fine: "SendMsg does not wait until the message is received by the client."